### ISSUE TITLE
RxJava 2.x artifact

### ISSUE THREAD

**Issue by ming13** [JakeWharton/RxReplayingShare#19]
Is it possible to publish a separate artifact for RxJava 2.x? We are currently migrating to RxJava 2.x and unable to use both `replayingShare` versions because dependency is the same. 

RxBinding for example follows [the policy you’ve described some time ago](http://jakewharton.com/java-interoperability-policy-for-major-version-updates/) and it is very helpful!

**Comment by JakeWharton** [comment]
This library is super tiny so I was hoping people using both would just copy it in. Don't you think it would be a bit weird to have two 2.0s if I were to make another release?

**Comment by ming13** [comment]
Yep, the size argument is totally valid and we did so eventually, but it has some disadvantages.

* We copied not only the operator, but the Kotlin extension as well, plus the tests. As we found out the `1.0.1` tag does not pass tests with latest RxJava and RxRelay versions. Replacing `PublishRelay` with `PublishSubject` allowed tests to pass but still, this is an extra work for someone.
* We put it as a separate module to isolate it from our code. This step is optional of course, but not everyone has multi-module project setup.
* Personally not a big fan of code copying since we have dependency management as a tool.

Anyway, we solved the issue for ourselves, but it still can be an issue for someone who is migrating to RxJava 2 and wants to have version-specific dependencies in parallel. Basically it is up to you ;-)
